### PR TITLE
Modifications to Content store

### PR DESCRIPTION
**Pull request by iTrooz**
I think the commit names already explain the changelog

### CODE REVIEW

**Comment by iTrooz** [comment]
https://github.com/WerWolv/ImHex/issues/794

File: plugins/builtin/include/content/views/view_store.hpp [hunk 1]
```diff
@@ -13,6 +13,13 @@
 
 namespace hex::plugin::builtin {
 
+    enum class RequestStatus {
+        NotAttempted,
+        Inprogress,
```
Comment (WerWolv) [inline plugins/builtin/include/content/views/view_store.hpp:18]: `InProgress`

File: plugins/builtin/source/content/views/view_store.cpp [hunk 2]
```diff
@@ -25,10 +25,10 @@ namespace hex::plugin::builtin {
     using namespace std::literals::chrono_literals;
 
     ViewStore::ViewStore() : View("hex.builtin.view.store.name") {
-        this->refresh();
-
         ContentRegistry::Interface::addMenuItem("hex.builtin.menu.help", 3000, [&, this] {
             if (ImGui::MenuItem("hex.builtin.view.store.name"_lang)) {
+                if(this->m_requestStatus==RequestStatus::NotAttempted)
```
Comment (WerWolv) [inline plugins/builtin/source/content/views/view_store.cpp:30]: Missing spaces before and after `==`

File: plugins/builtin/source/content/views/view_store.cpp [hunk 3]
```diff
@@ -126,6 +126,10 @@ namespace hex::plugin::builtin {
     }
 
     void ViewStore::refresh() {
+        // do not refresh if a refresh is already in progress
+        if(this->m_requestStatus == RequestStatus::Inprogress)return;
```
Comment (WerWolv) [inline plugins/builtin/source/content/views/view_store.cpp:130]: Missing space between `if` and `(`. `return` can go on a new line

File: plugins/builtin/source/content/views/view_store.cpp [hunk 4]
```diff
@@ -198,7 +203,10 @@ namespace hex::plugin::builtin {
                     this->parseResponse();
             }
 
-            this->drawStore();
+            if(this->m_requestStatus==RequestStatus::Failed)
+                ImGui::TextFormattedColored(ImGui::GetCustomColorVec4(ImGuiCustomCol_ToolbarRed), "Request failed !");
```
Comment (WerWolv) [inline plugins/builtin/source/content/views/view_store.cpp:207]: Remove the space before the `!`. This should probably also go into the localization files

**Review by WerWolv — CHANGES_REQUESTED** [review]

**Comment by iTrooz** [comment]
Done !

**Pull request merged**